src/app.js

```diff
@@ -0,0 +1,247 @@
+            funcName: "{that}.changeStarted",
+            args: [true]
+        }
+    },
+    modelListeners: {
+        keyedInSet: [{
+            funcName: "gpii.app.keyOut",
+            args: ["{change}.oldValue"]
+        }, {
+            funcName: "gpii.app.keyIn",
+            args: ["{change}.value"],
+            priority: "after:keyOut"
+        }]
+    },
+    invokers: {
+        changeStarted: {
+            changePath: "gpiiStarted",
+            value: "{arguments}.0"
+        }
+    }
+});
+
+gpii.app.startLocalFlowManager = function (that) {
+    var fluid = require("universal"),
+        gpii = fluid.registerNamespace("gpii");
+
+    if (os.platform() === "win32") {
+        require("gpii-windows/index.js");
+    }
+
+    gpii.start();
+    //TODO: needs to fire an on ready so we can be certain that everything has started
+    that.events.onAppReady.fire();
+};
+
+gpii.app.keyIn = function (token) {
+    request("http://localhost:8081/user/" + token + "/login", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+gpii.app.keyOut = function (token) {
+    request("http://localhost:8081/user/" + token + "/logout", function (/*error, response, body*/) {
+        //TODO Put in some error logging
+    });
+};
+
+/*
+ ** Component to create and update the task tray menu.
+ */
+fluid.defaults("gpii.taskTray.menu", {
+    gradeNames: "fluid.modelComponent",
+    menuLabels: {
+        tooltip: "GPII Electron",
+        keyedIn: "Keyed in with %setName", // string template
+        keyOut: "Key out %setName", //string template
+        notKeyedIn: "Not keyed in",
+        exit: "Exit",
+        keyIn: "Key in set ..."
+    },
+    snapsets: {
+        alice: "Alice",
+        davey: "Davey",
+        david: "David",
+        elaine: "Elaine",
+        elmer: "Elmer",
+        elod: "Elod",
+        livia: "Livia"
+    }
+});
+
+gpii.taskTray.updateSet = function (menu, menuLabels, setName, changeSetFn) {
+    if (setName) {
+        menu.push({ label: fluid.stringTemplate(menuLabels.keyedIn, {"setName": setName}), enabled: false });
+        menu.push({ label: fluid.stringTemplate(menuLabels.keyOut, {"setName": setName}),
+            click: function () {
+                changeSetFn("");
+            }
+        });
+    } else {
+        menu.push({ label: menuLabels.notKeyedIn, enabled: false });
+    }
+    return menu;
+};
+
+gpii.taskTray.updateSnapsets = function (menu, keyInLabel, snapsets, keyInFn) {
+    var submenuArray = [];
+    fluid.each(snapsets, function (value, key) {
+        submenuArray.push({label: value, click: function () { keyInFn(key); }});
+    });
+
+    menu.push({
+        label: keyInLabel,
+        submenu: submenuArray
+    });
+    return menu;
+};
+
+gpii.taskTray.addExit = function (menu, exitLabel) {
+    menu.push({
+        label: exitLabel,
+        click: function () {
+            var app = require("electron").app;
+            app.quit();
```

Comment:
Recall earlier feedback that this needs to fire an event which may cause a quit, rather than be hardwired in the menu structure. Without this, your code will be impossible to test. In general, the hard-wired click functions are a bad idea. I suggest that instead you provide each of them with a standard "click relay" which rather than including an implementation, just forward to a generic "menuItemClicked" event with appropriate arguments - or at the very least some component event specific to the menu item.